### PR TITLE
feat(subdomain): use subdomain by workspace

### LINKED ISSUE #8386: Environment variable IS_SIGN_UP_DISABLED ignored
## Description

Twenty ignores set environment variable IS_SIGN_UP_DISABLED with value "true". Twenty fails to block new workspace creation.

## Expected behavior

Block the creation of new accounts and workspaces without explicit invites.

## Details

Install command: bash <(curl -sL https://git.new/20)
Reproduced with: twenty v0.31.0 – v0.32.0
Fresh install: Yes

### PR DESCRIPTION
**Pull request by AMoreaux**
Resolve #8386 

-   [ ] Ensure SSO is functioning properly
-   [ ] Develop a strategy for feature flags and deployment that minimizes downtime and risk
-   [ ] Add documentation for environment variables
-   [ ] Enable listing or creation of workspaces upon logging into the app with Google Connect
-   [ ] Update the cookie in the new domain when a workspace is renamed
-   [x] Automatically connect to the default or first workspace without requiring workspace selection
-   [ ] Swap the positions of the twenty and workspace logos, and make the twenty logo clickable to redirect to the homepage
-   [x] Implement captchas for public routes. Check Guard Captcha for integration

### CODE REVIEW

**Review by FelixMalfait — COMMENTED** [review]
I started having a quick look but not in depth. Looks like great work!!! Big piece!

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/hooks/useWorkspaceSwitching.ts [hunk 1]
```diff
@@ -1,74 +1,29 @@
...
-} from '@/auth/states/signInUpStepState';
-import { tokenPairState } from '@/auth/states/tokenPairState';
-import { AppPath } from '@/types/AppPath';
-import { useGenerateJwtMutation } from '~/generated/graphql';
+import { useSwitchWorkspaceMutation } from '~/generated/graphql';
 import { isDefined } from '~/utils/isDefined';
-import { sleep } from '~/utils/sleep';
+import { buildWorkspaceUrl } from '~/utils/workspace-url.helper';
 
 export const useWorkspaceSwitching = () => {
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/ui/navigation/navigation-drawer/hooks/useWorkspaceSwitching.ts:9]: The workspace switcher still works locally even though I'm in mono-domain mode, is that expected? I thought we wouldn't manage to make it work without multi-domain

File: packages/twenty-server/src/engine/utils/get-workspace-subdomain-by-origin.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const getWorkspaceSubdomainByOrigin = (origin: string) => {
```
[the remark this answers is not in the document]
Comment (AMoreaux) [reply packages/twenty-server/src/engine/utils/get-workspace-subdomain-by-origin.ts:1]: It's from the origin header:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

File: packages/twenty-server/src/database/typeorm/core/migrations/1730137590546-addSubdomainToWorkspace.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  implements MigrationInterface
+{
+  name = 'AddSubdomainToWorkspace1730137590546';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "core"."workspace" ADD "subdomain" varchar NULL`,
+    );
+    await queryRunner.query(
+      `UPDATE "core"."workspace" SET "subdomain" = LOWER(REPLACE("id", '-', ''))`,
```
[the remark this answers is not in the document]
Comment (AMoreaux) [reply packages/twenty-server/src/database/typeorm/core/migrations/1730137590546-addSubdomainToWorkspace.ts:13]: `LOWER` and `REPLACE` are not really necessary so I removed them.

File: packages/twenty-server/test/utils/createTestingService.utils.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        useValue: override?.services?.[name] ?? {},
+      })) as Provider[]),
+    ],
+  }).compile();
+
+  // @ts-ignore
+  return module.get<T>(service);
+};
+
+createTestingService(AuthService);
```
[the remark this answers is not in the document]
Comment (AMoreaux) [reply packages/twenty-server/test/utils/createTestingService.utils.ts]: This file didn't exist anymore, it was a test to improve testing but it's another subject.

**Comment by AMoreaux** [comment]
Please do not review this PR, I will split it into smaller PR.

**Pull request closed without merging**